### DISCUSSION TITLE
in chi kar mikone ?

### DISCUSSION THREAD

**Discussion by amirfoad** [vestaabner/Channel#2] [General]
in chi kar mikone ?

**Comment by AhmadrezaHamidi** [comment]
این مقاله توضیح روشنی از اینکه همزمانی و موازی بودن چیست و چگونه می توان از کانال ها برای تسهیل ارتباط و همگام سازی بین رشته ها استفاده کرد، ارائه می دهد.
همزمانی به توانایی یک برنامه برای انجام چندین کار به طور همزمان اشاره دارد، در حالی که موازی سازی به اجرای همزمان چندین کار است. همزمانی موازی‌سازی را امکان‌پذیر می‌سازد و یک برنامه همزمان ممکن است از موازی‌سازی سود ببرد، اما هدف آن داشتن یک ساختار خوب است. کانال ها راهی برای برقراری ارتباط و همگام سازی بین رشته ها، بدون نیاز به همگام سازی خارجی یا مکانیسم های قفل ارائه می کنند.
در دات نت می توانیم با استفاده از کلاس Channel کانال هایی ایجاد کنیم و سپس با استفاده از رایتر و ریدر کانال از کانال بنویسیم و بخوانیم. کانال ها می توانند محدود یا بدون محدودیت باشند و هر دو بین رشته ها ارتباط برقرار می کنند و همگام می شوند.
این مقاله همچنین چندین مثال از تکنیک‌های برنامه‌نویسی همزمان برای کار با کانال‌ها، از جمله الگوی ژنراتور، الگوی فن‌آوری و الگوی خط لوله ارائه می‌دهد. این الگوها نشان می‌دهند که چگونه می‌توان از کانال‌ها برای ایجاد جریان‌های کاری همزمان که کارآمد و مقیاس‌پذیر هستند استفاده کرد.
به طور کلی، این مقاله مقدمه مفیدی برای هماهنگ سازی ساختارهای داده در فضای نام System.Threading.Channels دات نت ارائه می دهد و نشان می دهد که چگونه می توان از آنها برای طراحی گردش کار همزمان استفاده کرد.    
از جاهایی که این کد کاربرد دارد میتوان این را مثال زد که فرض کنیم که Api Like در instagram  بیشتر از بقیه ی Api در حال صدا شدن هست . و ممکن است به خاطر Request های زیاد اپیکیشن دان شود 
یکی از کار هایی که میشه کرد اینه که ما یک صف برای این Request ها ایجاد کرده و وقتی که request به Api میرسد ما RESULTI که این پست لایک شده در برمیگردانیم ولی در واقع این ریکوست وارد صف شده ، و کامپایلر  هر کدام از این ریکوست ها را برداشته و روی دیتای مورد نظرش تغییرات اعمال میکند . 

https://github.com/vestaabner/Channel
